Source/Particles/WarpXParticleContainer.cpp

```diff
@@ -534,6 +538,11 @@ WarpXParticleContainer::DepositCharge (Vector<std::unique_ptr<MultiFab> >& rho,
 #ifdef _OPENMP
         }
 #endif
+
+#ifdef WARPX_DIM_RZ
+        if (reset) WarpX::GetInstance().ApplyInverseVolumeScalingToChargeDensity(rho[lev].get(), lev);
+#endif
```

Comment:
It seems that, for RZ:
- If `reset`, `WarpXParticleContainer::DepositCharge` applies the inverse volume scaling
- If `not reset`, this function does not apply the volume scaling, meaning that the volume scaling has to be applied after this function call.

I understand that this PR is not meant to support RZ, so do you plan to modify this behavior in a subsequent PR? For instance, `WarpXParticleContainer::DepositCharge` could take an additional argument `if_rz_do_volume_scaling`, so that it is more explicit (though not very nice).

Reply:
Yes, that's a good point ; I'll add this.